### DISCUSSION TITLE
Is it possible to define an object in your schema with unknown/dynamic keys?

### DISCUSSION THREAD

**Discussion by alexedwardjones** [sensedeep/dynamodb-onetable#428] [Q&A] [answered]
**Summary**

I'm working on a project where I want to store my data shaped like this example...
```
{
  name: "Bob",
  books: {
     bookName1: {
       name: "BookName1",
       pages: 123
     },
     bookName2: {
       name: "BookName2",
       pages: 456
     },
  }
}
```
So instead of using an array or a separate model, the books are an object where they properties are the book names.

I don't know what the keys of the `books` object would be but I do know the structure of each individual `book` object. Is there a way to define this in the schema?

**Comment by mobsense** [comment, marked as answer]
If you don't know the key, you can use a generated ID and then fetch based on ID and filter on name. That works okay for a small number of books.

A separate model is the way to go. If the book name is unique, use that in the key:

```
models: {
    Book: {
        pk:               { type: 'string', value: 'book#${name}' },
        sk:               { type: 'string', value: 'book#' },
        name:         { type: 'string' }
        pages:        { type: 'number', default: 0 },
      

```
